src/v2/Apps/Show/Components/ShowPartnerEntityHeader.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react"
+import { ContextModule, Intent } from "@artsy/cohesion"
+import { createFragmentContainer, graphql } from "react-relay"
+import { Clickable, EntityHeader, Text } from "@artsy/palette"
+import { useSystemContext } from "v2/Artsy"
+import { filterLocations } from "v2/Apps/Artwork/Utils/filterLocations"
+import { limitWithCount } from "v2/Apps/Artwork/Utils/limitWithCount"
+import { FollowProfileButtonFragmentContainer as FollowProfileButton } from "v2/Components/FollowButton/FollowProfileButton"
+import { ShowPartnerEntityHeader_partner } from "v2/__generated__/ShowPartnerEntityHeader_partner.graphql"
+import { openAuthToFollowSave } from "v2/Utils/openAuthModal"
+
+interface ShowPartnerEntityHeaderProps {
+  partner: ShowPartnerEntityHeader_partner
+}
+
+const ShowPartnerEntityHeader: React.FC<ShowPartnerEntityHeaderProps> = ({
+  partner,
+}) => {
+  const { user, mediator } = useSystemContext()
+
+  if (!partner) {
+    return null
+  }
+
+  const imageUrl = partner.profile?.icon?.url
+  const partnerInitials = partner.initials
+  const canFollow =
+    partner && partner.type !== "Auction House" && !!partner.profile
+  const locationNames = limitWithCount(
+    filterLocations(partner.locations),
+    2
+  ).join(", ")
```

Comment:
All this is largely copied from another artwork page component — though I am completely horrified by the display logic and how complex it is to just setup a lil button with a follow.

Reply:
I wonder what behavior (these days), is triggered or affected by following a gallery profile. Like, why can't you follow an auction house?